Foundation/NSOrderedSet.swift

```diff
@@ -37,38 +48,151 @@ public class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     
     public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
     
-    public var count: Int { NSUnimplemented() }
-    public func objectAtIndex(idx: Int) -> AnyObject { NSUnimplemented() }
-    public func indexOfObject(object: AnyObject) -> Int { NSUnimplemented() }
-    public override init() { NSUnimplemented() }
-    public init(objects: UnsafePointer<AnyObject?>, count cnt: Int) { NSUnimplemented() }
+    public var count: Int {
+        return _entries.count
+    }
+
+    public func objectAtIndex(idx: Int) -> AnyObject {
+        return _entryAtIndex(idx).object
```

Comment:
That's right, except that using an array as a backing store is more costly when it comes to insertion and deletion (O(n) in most cases). The proposal I have guarantees O(1) performance for everything except getting an item by its index.

The trade-off to consider here is whether we want to optimize for accessing by index, at the cost of expensive mutation, or optimize for mutation at the cost of accessing by index.

I'm happy to go either route. I don't feel like I have enough data to state which case makes most sense to optimize for.